rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py

```diff
@@ -1,66 +1,84 @@
+from __future__ import annotations
 import logging
-from collections import defaultdict, OrderedDict
-from pathlib import Path
+from collections import OrderedDict
 
+import scipy.sparse
 import numpy as np
 from typing import (
     Any,
     Dict,
-    Optional,
     Text,
     List,
-    Type,
-    Union,
+    Tuple,
+    Hashable,
     Callable,
+    Set,
+    Optional,
 )
 
-from rasa.nlu.tokenizers.spacy_tokenizer import POS_TAG_KEY
-from rasa.shared.constants import DOCS_URL_COMPONENTS
-from rasa.nlu.components import Component
+from rasa.engine.graph import ExecutionContext, GraphComponent
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.nlu.tokenizers.spacy_tokenizer import POS_TAG_KEY, SpacyTokenizer
 from rasa.nlu.tokenizers.tokenizer import Token
 from rasa.nlu.tokenizers.tokenizer import Tokenizer
-from rasa.nlu.featurizers.featurizer import SparseFeaturizer
-from rasa.shared.nlu.training_data.features import Features
-from rasa.nlu.config import RasaNLUModelConfig
+from rasa.nlu.featurizers.sparse_featurizer.sparse_featurizer import SparseFeaturizer2
+from rasa.nlu.constants import TOKENS_NAMES
+from rasa.shared.constants import DOCS_URL_COMPONENTS
 from rasa.shared.nlu.training_data.training_data import TrainingData
 from rasa.shared.nlu.training_data.message import Message
-from rasa.nlu.constants import TOKENS_NAMES, FEATURIZER_CLASS_ALIAS
-from rasa.shared.nlu.constants import TEXT, FEATURE_TYPE_SEQUENCE
-
-from rasa.nlu.model import Metadata
-import rasa.utils.io as io_utils
+from rasa.shared.nlu.constants import TEXT
+from rasa.shared.exceptions import InvalidConfigException
+import rasa.shared.utils.io
+import rasa.utils.io
+from rasa.nlu.featurizers.sparse_featurizer._lexical_syntactic_featurizer import (
+    LexicalSyntacticFeaturizer,
+)
 
 logger = logging.getLogger(__name__)
 
-END_OF_SENTENCE = "EOS"
-BEGIN_OF_SENTENCE = "BOS"
+# TODO: remove after all references to old featurizer have been removed
+LexicalSyntacticFeaturizer = LexicalSyntacticFeaturizer
 
 
-class LexicalSyntacticFeaturizer(SparseFeaturizer):
-    """Creates features for entity extraction.
+END_OF_SENTENCE = "EOS"
+BEGIN_OF_SENTENCE = "BOS"
 
-    Moves with a sliding window over every token in the user message and creates
-    features according to the configuration.
+FEATURES = "features"
+
+
+class LexicalSyntacticFeaturizerGraphComponent(SparseFeaturizer2, GraphComponent):
+    """Extracts and encodes lexical syntactic features.
```

Comment:
thanks for these amazing docstrings 🙌🏻 🚀

Reply:
... with lots of ` instead of ' :D